### ISSUE TITLE
AndroidSupportedAbis problem

### ISSUE THREAD

**Issue by ptsiogas** [praeclarum/sqlite-net#495]
The latest version (1.2.1) makes my project support the x86_64 abi even if i haven't selected it. I figured this out when i tried to upload an .apk update to google play console.
Can i exclude it somehow?

**Comment by ericsink** [comment]
What is the specific error message you are seeing?

**Comment by ptsiogas** [comment]
It's not an error, it's the fact that i want to support only **armeabi, armeabi-v7a, x86** and the lib forces me to support **x86_64** too.
I don't want to do that because Xamarin specifically says that x64 is still in experimental stage:
> Note: 64-bit runtime support is currently an experimental feature in Xamarin.Android 5.1.x. Remember that 64-bit runtimes are not required in order to run your app on 64-bit devices.

**Comment by ericsink** [comment]
Ah.

I'm still trying to figure out more precisely what you mean when you say the lib is forcing you to support x86_64.

It is true that SQLitePCL.raw includes a native lib for x86_64.  Are you saying that you are seeing that file is still in the apk even though you didn't choose that abi in your project options?

**Comment by ptsiogas** [comment]
Yes that's exactly what's happening.
When you upload an .apk to the google play store, there is a field called "Native platforms" which is displaying the supported abis. My .config file includes only **armeabi, armeabi-v7a, x86** but the "Native platforms" fields shows also the **x86_64** abi. I tried to downgrade your lib to 1.0.11 and the **x86_64** abi removed.

**Comment by ericsink** [comment]
The nuget package SQLitePCLRaw.lib.e_sqlite3.android contains an assembly which contains several native sqlite3 libraries as embedded resources.  One of those libraries is for x86_64, so that projects using SQLitePCL.raw can support x86_64 if they want to.

The Xamarin tooling extracts those library files from the resources and places them in the apk.  I would assume that it should only copy the libraries needed for the native platforms you have selected in your project options.  In other words, if you do not select x86_64 in your project options, it should not copy the x86_64 library into your apk.  Or at least that is how I assume things would work.

Neither SQLitePCL.raw nor sqlite-net-pcl (AFAIK) do anything to change the configuration of your project options to change which native platforms you selected.

I do not know how the "Native platforms" field you mentioned gets its value.  It is apparently not based on the values in your config file.  Is it including x86_64 in the list simply because it sees an x86_64 library in the apk?

Bottom line:  I still don't know where in the build pipeline this is going wrong.

**Comment by praeclarum** [comment]
@ptsiogas Sorry but this seems to be more of a limitation of Xamarin than this library. Good luck!
